crates/testlang/outputs/cargo/tests/src/query/parser_tests.rs

```diff
@@ -55,9 +55,65 @@ fn test_parsing_error() {
     let result = Query::parse(r#"@root [_ ..."#);
     match result {
         Ok(_) => panic!("Expected error"),
-        Err(e) => assert_eq!(
+        Err(e) => {
+            assert_eq!(
+                e.message,
+                "Parse error:\nexpected ']' at: \nAlt at: [_ ...\nAlt at: @root [_ ...\n"
+            );
+            assert_eq!((e.row, e.column), (0, 12));
+        }
+    }
+}
+
+// See https://github.com/NomicFoundation/slang/issues/1042
+#[test]
+fn test_parsing_error_with_invalid_edge_label() {
+    let result = Query::parse(
+        r#"
+[Tree
+    ...
+    @name Name: [_]
+    ...
+]"#,
+    );
+    match result {
+        Ok(_) => panic!("Expected error"),
+        Err(e) => {
+            assert_eq!(
+                e.message,
+                "Parse error:\nMapRes at: Name: [_]\n    ...\n]\nin section 'parsing edge label', at: Name: [_]\n    ...\n]\n",
```

Comment:
In the following error, the inner message here is actually `MapRes`. Is it possible to change it to say something about the specific error? for example `'@name' is not a valid EdgeLabel.`

```log
Parse error:
    MapRes at: Name: [_]
        ...
    ]
    in section 'parsing edge label', at: Name: [_]
        ...
    ]
```

Reply:
Same suggestion for terminals/non-terminals.

Reply:
Same suggestion for node kinds.

Reply:
This will require implementing and using a custom error type throughout the parser. I'm gonna work on this as it keeps coming up.

Reply:
:heavy_check_mark: